lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3781,15 +3802,15 @@ void TypeChecker::validateDecl(ValueDecl *D) {
     // Check generic parameters, if needed.
     DeclValidationRAII IBV(typeAlias);
 
-    validateGenericTypeSignature(typeAlias);
+    (void)typeAlias->getGenericSignature();
```

Comment:
This shouldn't be called from validateDecl(). You should force the request in typeCheckDecl() (or rather DeclChecker::visitTypeAliasDecl(), etc, instead)